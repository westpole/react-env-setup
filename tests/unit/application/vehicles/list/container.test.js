import toJson from 'enzyme-to-json';

import Container from '../../../../../src/scripts/application/vehicles/list/container';
import { mockWithStore } from '../../../../tools/mock-component.jsx';

describe('Application/Vehicles/List/Container', () => {
  it('should match saved snapshot', () => {
    const wrapper = mockWithStore({
      type: 'shallow',
      scope: Container,
      initialState: {
        vehicles: [],
      },
    });

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should connect Container to the Store', () => {
    const wrapper = mockWithStore({
      type: 'shallow',
      scope: Container,
      initialState: {
        vehicles: [],
      },
    });

    expect(wrapper.instance().store).toBeDefined();
  });

  it('should pass Properties to the Component', () => {
    const wrapper = mockWithStore({
      type: 'shallow',
      scope: Container,
      initialState: {
        vehicles: [],
      },
    });

    expect(wrapper.props()).toEqual({
      dispatch: expect.any(Function),
      vehicles: [],
    });
  });
});
