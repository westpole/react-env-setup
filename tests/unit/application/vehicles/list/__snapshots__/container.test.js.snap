// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application/Vehicles/List/Container should match saved snapshot 1`] = `
<Component
  dispatch={[Function]}
  vehicles={Array []}
/>
`;
